datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -537,8 +537,15 @@ fn type_union_resolution_coercion(
         }
         (DataType::Dictionary(index_type, value_type), other_type)
         | (other_type, DataType::Dictionary(index_type, value_type)) => {
-            let new_value_type = type_union_resolution_coercion(value_type, other_type);
-            new_value_type.map(|t| DataType::Dictionary(index_type.clone(), Box::new(t)))
+            match type_union_resolution_coercion(value_type, other_type) {
+                // Dict(k, Utf8View) is redundant, Utf8View is good enough
+                Some(DataType::Utf8View) => Some(DataType::Utf8View),
```

Comment:
Possibly also true for binary views (even if I didn't report in my original issue)?